tool/tbot/config/configtemplate_ssh.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/tool/tbot/destination"
+	"github.com/gravitational/teleport/tool/tbot/identity"
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+)
+
+// TemplateSSHClient contains parameters for the ssh_config config
+// template
+type TemplateSSHClient struct {
+	ProxyPort uint16 `yaml:"proxy_port"`
+}
+
+func (c *TemplateSSHClient) CheckAndSetDefaults() error {
+	if c.ProxyPort == 0 {
+		c.ProxyPort = defaults.SSHProxyListenPort
+	}
+	return nil
+}
+
+func (c *TemplateSSHClient) Describe() []FileDescription {
+	return []FileDescription{
+		{
+			Name:     "ssh_config",
+			ModeHint: destination.ModeHintSecret,
+		},
+		{
+			Name:     "known_hosts",
+			ModeHint: destination.ModeHintSecret,
+		},
+	}
+}
+
+func (c *TemplateSSHClient) Render(authClient *auth.Client, currentIdentity *identity.Identity, destination *DestinationConfig) error {
+	if !destination.ContainsKind(KindSSH) {
+		return trace.BadParameter("%s config template requires kind `ssh` to be enabled", TemplateSSHClientName)
+	}
+
+	dest, err := destination.GetDestination()
+	if err != nil {
+		return err
+	}
+
+	var (
+		proxyHosts     []string
+		firstProxyHost string
+		firstProxyPort string
+	)
+
+	clusterName, err := authClient.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	proxies, err := authClient.GetProxies()
```

Comment:
You may be able to get this info easier from `authClient.Ping()`. In theory I think all proxies should have the same public addr

Reply:
Nice, that's a lot simpler, thanks!